### DISCUSSION TITLE
Adding custom menu options in the custom created top-panel.

### DISCUSSION THREAD

**Discussion by Abhay0294** [eclipse-theia/theia#11211] [Q&A]
How can we add custom menu options to the custom Theia top panel[The image of which is attached in the snapshot]

<img width="837" alt="CustomPanel_MenuOption" src="https://user-images.githubusercontent.com/57217189/170648513-34324ea5-370e-4274-8b50-c90ec5130002.PNG">

Things I tried:
I am able to add the custom menu options in the default menu bar.

But my requirement is to add a Home button with the action attached which should be present in the custom top-panel, which is pointed out in the snapshot.

Requesting necessary suggestions in this case from the community.

**Comment by colin-grant-work** [comment]
It sounds like the best solution would be to replicate approximately the code that attaches the menu items to the main menu. In the browser, that means creating and populating a `MenuBarWidget` and attaching it in your new node. The model code in Theia is in [browser-menu-plugin.ts](https://github.com/eclipse-theia/theia/blob/master/packages/core/src/browser/menu/browser-menu-plugin.ts).

**Reply by Abhay0294** [reply]
Thanks @colin-grant-work , for the suggestion.  Just one thing I want to ask that, can **browser-menu-plugin.ts**  file can be overrided to create the custom functionality to support the requirement ?
